### PR TITLE
Adding manifest parameter to 'redshift.copy_from_files' method

### PR DESCRIPTION
**Pull request by malachi-constant**
Subject: Adding manifest parameter to 'copy_from_files 'method

### Feature
- Feature

### Detail
- New parameter for `redshift.copy_from_files(...)` : `manifest`
- New method `redshift._get_paths_from_manifest(...)`: Returns list of S3 paths declared in manifest file.

### Relates
- [#1142 ](https://github.com/awslabs/aws-data-wrangler/issues/1142)

- By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### Validation 
- Unit test written for method using new parameter.
```
> pytest tests/test_redshift.py::test_copy_from_files_manifest
============================================================================== test session starts ===============================================================================
platform darwin -- Python 3.7.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /Users/hansonlu/work/awslabs/code/github/aws-data-wrangler/.venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/hansonlu/work/awslabs/code/github/aws-data-wrangler, configfile: pyproject.toml
plugins: xdist-2.5.0, anyio-3.5.0, forked-1.4.0, timeout-2.1.0, rerunfailures-10.2, cov-3.0.0
collected 1 item

tests/test_redshift.py::test_copy_from_files_manifest PASSED                                                                                                               [100%]

=============================================================================== 1 passed in 5.14s ================================================================================
```

- Unit test written for new method: `_get_paths_from_manifest(...)`
```
> pytest tests/test_redshift.py::test_get_paths_from_manifest
============================================================================== test session starts ===============================================================================
platform darwin -- Python 3.7.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /Users/hansonlu/work/awslabs/code/github/aws-data-wrangler/.venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/hansonlu/work/awslabs/code/github/aws-data-wrangler, configfile: pyproject.toml
plugins: xdist-2.5.0, anyio-3.5.0, forked-1.4.0, timeout-2.1.0, rerunfailures-10.2, cov-3.0.0
collected 1 item

tests/test_redshift.py::test_get_paths_from_manifest PASSED                                                                                                                [100%]

=============================================================================== 1 passed in 1.66s ================================================================================

```

### CODE REVIEW

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 2ad9114b6bbc37e049da43ec7de6eee6f11d60dc
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d188fe8d-4c0f-47d3-8dc4-e852566f2e1d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
One point I was unsure on, is `path` still required for the method even when using a `manifest` uri ?

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: f4e7d5873c293936c4db7f77d9d0a584aaa4effb
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=186e7b56-783f-4e35-8512-6293630f2296%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by kukushking** [comment]
@malachi-constant looks like it's [either path or manifest](https://docs.aws.amazon.com/redshift/latest/dg/copy-parameters-data-source-s3.html#copy-parameters-data-source-s3-syntax)

**Comment by kukushking** [comment]
Just a thought - maybe manifest could be a boolean param that would determine if path is just an S3 path or path to the manifest? Otherwise making path optional would work just as well since it wouldn't be a breaking change

**Review by kukushking — APPROVED** [review]

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 40638b72ddab998de155854f9114c0d93739a7ed
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cf198574-788b-4f6c-aac8-dd24f860a431%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 9261ae6d5eef954ed25e1678e7b93e729624f076
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=65b8f576-60df-45ea-94c3-a84152c35915%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 726c27002367f3abf79db3c9555969fcedfd5d7e
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=57115724-3845-457b-bed1-a9b4753817a6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]

**Review by jaidisido — COMMENTED** [review]
Looks great, left minor comments

File: awswrangler/redshift.py [hunk 1]
```diff
@@ -302,6 +317,17 @@ def _create_table(  # pylint: disable=too-many-locals,too-many-arguments
...
         )
     elif path is not None:
+        if manifest:
+            if isinstance(path, str):
+                path = _get_paths_from_manifest(
+                    path=path,
+                    boto3_session=boto3_session,
+                )
+            else:
+                raise TypeError(
```
[the remark this answers is not in the document]
Comment (malachi-constant) [reply awswrangler/redshift.py:327]: `mypy` doesn't like that we could pass str or list to that method.

```
+ mypy --install-types --non-interactive awswrangler
awswrangler/redshift.py:327: error: Argument "path" to "_get_paths_from_manifest" has incompatible type "Union[str, List[str]]"; expected "str"
Found 1 error in 1 file (checked 71 source files)
```

Should I allow a list type as well for `path` in that method to catch that case ??

[the remark this answers is not in the document]
Comment (jaidisido) [reply awswrangler/redshift.py:327]: Nah it's fine the way it's 👍🏼

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 80cdaad2d68b8392d17ed7b8d41dd6667ecd0ad1
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=161dea92-1ab4-4f0c-96ee-f4f0dec8b804%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**